client/src/rpc_client.rs

```diff
@@ -2936,6 +2964,90 @@ impl RpcClient {
         }
     }
 
+    pub fn get_latest_blockhash(&self) -> ClientResult<Hash> {
+        let (blockhash, _) = self.get_latest_blockhash_with_commitment(self.commitment())?;
+        Ok(blockhash)
+    }
+
+    pub fn get_latest_blockhash_with_commitment(
+        &self,
+        commitment: CommitmentConfig,
+    ) -> ClientResult<(Hash, u64)> {
+        let latest_blockhash = self
+            .send::<Response<RpcBlockhash>>(
+                RpcRequest::GetLatestBlockhash,
+                json!([self.maybe_map_commitment(commitment)?]),
+            )?
+            .value;
+
+        let blockhash = latest_blockhash.blockhash.parse().map_err(|_| {
+            ClientError::new_with_request(
+                RpcError::ParseError("Hash".to_string()).into(),
+                RpcRequest::GetLatestBlockhash,
+            )
+        })?;
+        Ok((blockhash, latest_blockhash.last_valid_block_height))
+    }
+
+    pub fn is_blockhash_valid(
+        &self,
+        blockhash: &Hash,
+        commitment: CommitmentConfig,
+    ) -> ClientResult<bool> {
+        let result = self.send::<Response<bool>>(
+            RpcRequest::IsBlockhashValid,
+            json!([blockhash.to_string(), commitment,]),
+        )?;
+        Ok(result.value)
+    }
+
+    pub fn get_fee_for_transaction(
```

Comment:
heh, good point, I like that it takes a message, fee does not depend on payer or recent_blockhash.  I'm fine with `get_fee_for_message`

Reply:
Actually, lets stick with "Transaction", I'll change the parameters

Reply:
I'm oscillating, I'm going with `get_fee_for_message`

Reply:
I know what you mean. Even though it's a little weird for between-crate consistency, I kind of like going with `get_fee_for_transaction` in rpc (since send and simulate already take serialized Transactions), and `get_fee_for_message` in runtime/client/etc

Reply:
I'm adverse to sending the useless signature data over rpc.  More efficient to just send the message.  Good enough reason to go with `get_fee_for_message`?

Reply:
Fair point, but all we have at the moment for deser and sanitization is this, which expects a serialized Transaction: https://github.com/solana-labs/solana/blob/9dfeee2993312accb61b3d48270671de2774eb81/rpc/src/rpc.rs#L3879

Reply:
True, but not hard to go from tx to message or vis versa, could have the api take a tx but the otw rpc take a message...  Also, most callers seem to have a message and calc fees before signing.  I'm down either way, lob back to you 🙂

Reply:
That api takes a serialized tx and doesn't spit out a Transaction until nearly the end, so it's useless for `get_fee_for_message`. But that's fine, I'm on board with Message. We'll just have to duplicate some of that validation, is all.